File: scripts/deno.ts
import * as path from 'path';
import * as fs from 'fs';
import * as fsPromises from 'fs/promises';

function isDirectory(filePath: string): boolean {
  const fileStats = fs.lstatSync(filePath);
  return fileStats.isDirectory();
}

type ForTreePredicate = (file: string) => Promise<void>;

async function forTree(basePath: string, predicate: ForTreePredicate): Promise<void> {
  const files = await fsPromises.readdir(basePath);
  const fileTransformers: Promise<void>[] = [];
  const subdirectoryIterators: Promise<void>[] = [];
  files.forEach((file) => {
    const filePath = path.join(basePath, file);
    if (isDirectory(filePath)) {
      subdirectoryIterators.push(forTree(filePath, predicate));
    } else {
      fileTransformers.push(predicate(filePath));
    }
  });
  await Promise.all([...subdirectoryIterators, ...fileTransformers]);
}

console.log('Generating Deno friendly TypeScript files');

const SRC_PATH = './src';
const DENO_PATH = './deno';
const importLineEndRegex = /(?<=\S)';$/gm;

try {
  if (fs.existsSync(DENO_PATH) && isDirectory((DENO_PATH))) {
    fs.rmSync(DENO_PATH, { recursive: true });
  }
  await fsPromises.cp(SRC_PATH, DENO_PATH, { recursive: true });
  await forTree(DENO_PATH, async (filePath) => {
    const fileContent = await fsPromises.readFile(filePath, 'utf8');
    const transformedFileContent = fileContent.replace(importLineEndRegex, '.ts\';');
    await fsPromises.writeFile(filePath, transformedFileContent);
  });
} catch (error) {
  console.error(error);
  process.exit(-1);
}
console.log('Success!');

File: scripts/deno.py
import os
import shutil
import re

src_path = "./src"
deno_path = "./deno"
regex = r"';$"

if os.path.exists(deno_path):
    if os.path.isdir(deno_path):
        shutil.rmtree(deno_path)
    else:
        os.remove(deno_path)

shutil.copytree(src_path, deno_path)

for root, dirs, files in os.walk(deno_path):
    for file in files:
        file_path = os.path.join(root, file)
        lines = []
        with open(file_path, "r") as file_handle:
            lines = file_handle.readlines()
            for i in range(len(lines)):
                if lines[i].startswith('import'):
                    lines[i] = re.sub(regex, ".ts';", lines[i])

        with open(file_path, "w") as file_handle:
            file_handle.writelines(lines)
